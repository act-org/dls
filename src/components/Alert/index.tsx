/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

/* eslint-disable react/display-name */

import { AlertProps as MuiAlertProps } from '@mui/lab';
import { Alert as MuiAlert, AlertTitle as MuiAlertTitle } from '@mui/material';
import { FC, forwardRef, ReactElement } from 'react';

export type AlertProps = MuiAlertProps & {
  title?: string | ReactElement;
};

export const Alert: FC<AlertProps> = forwardRef(
  (
    { children, title, ...props }: AlertProps,
    ref,
  ): ReactElement<AlertProps> => {
    return (
      <MuiAlert ref={ref} {...props}>
        {title && <MuiAlertTitle>{title}</MuiAlertTitle>}

        {children}
      </MuiAlert>
    );
  },
);

Alert.defaultProps = {
  title: undefined,
};

export default Alert;
