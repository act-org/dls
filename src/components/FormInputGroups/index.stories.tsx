/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';
import { MenuItem, TextField } from '@material-ui/core';
import { Meta, Story } from '@storybook/react/types-6-0';

import FormInputGroups, { FormInputGroupsProps } from '.';

const Template: Story<FormInputGroupsProps> = args => {
  const [value1, setValue1] = React.useState('Value 1');
  const [value2, setValue2] = React.useState('Value 2');
  const [value3, setValue3] = React.useState('undefined');
  const [value4, setValue4] = React.useState('Value 4');
  const [value5, setValue5] = React.useState('Value 5');

  return (
    <FormInputGroups
      groups={[
        [
          {
            formInput: (
              <TextField
                label="Input 1"
                onChange={(e: any): void => {
                  setValue1(e.target.value);
                }}
                placeholder="Required Text"
                required
                type="text"
                value={value1}
              />
            ),
            key: 'INPUT_1',
          },
          {
            formInput: (
              <TextField
                label="Input 2"
                onChange={(e: any): void => {
                  setValue2(e.target.value);
                }}
                placeholder="Required Password"
                required
                type="password"
                value={value2}
              />
            ),
            key: 'INPUT_2',
          },
        ],
        [
          {
            formInput: (
              <TextField
                fullWidth
                label="Input 3"
                onChange={(e: any): void => {
                  setValue3(e.target.value);
                }}
                select
                type="password"
                value={value3}
              >
                <MenuItem key="undefined" value="undefined">
                  Select
                </MenuItem>

                <MenuItem key="ONE" value="ONE">
                  One
                </MenuItem>

                <MenuItem key="TWO" value="TWO">
                  Two
                </MenuItem>

                <MenuItem key="THREE" value="THREE">
                  Three
                </MenuItem>
              </TextField>
            ),
            key: 'INPUT_3',
          },
        ],
        [
          {
            formInput: (
              <TextField
                label="Input 4"
                onChange={(e: any): void => {
                  setValue4(e.target.value);
                }}
                placeholder="Required Number"
                required
                type="number"
                value={value4}
              />
            ),
            key: 'INPUT_4',
          },
          {
            formInput: (
              <TextField
                label="Input 5"
                onChange={(e: any): void => {
                  setValue5(e.target.value);
                }}
                placeholder="Optional Text"
                type="text"
                value={value5}
              />
            ),
            key: 'INPUT_5',
          },
        ],
      ]}
      {...args}
    />
  );
};

export const Preview: Story<FormInputGroupsProps> = Template.bind({});

export default {
  component: FormInputGroups,
  parameters: {
    info: {
      text: 'This is a custom FormInputGroups component.',
    },
  },
  title: 'Organisms/FormInputGroups',
} as Meta<FormInputGroupsProps>;
