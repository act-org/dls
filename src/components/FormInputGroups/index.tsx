/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';
import { Grid } from '@material-ui/core';

import useStyles from './styles';

interface FormInput {
  formInput: React.ReactElement<unknown>;
  key: string;
}

type FormInputGroup = FormInput[];

export interface FormInputGroupsProps {
  groups: FormInputGroup[];
}

export const FormInputGroups: React.FC<FormInputGroupsProps> = ({
  groups,
}: FormInputGroupsProps) => {
  const classes = useStyles();

  return (
    <>
      {/* eslint-disable react/no-array-index-key */}
      {groups.map(
        (group, i): React.ReactElement<unknown> => (
          <Grid
            classes={{
              container: classes.gridContainerRoot,
            }}
            container
            key={i}
            spacing={4}
          >
            {group.map(
              (input): React.ReactElement<unknown> => (
                <Grid
                  classes={{
                    item: classes.gridItemRoot,
                  }}
                  item
                  key={input.key}
                >
                  {input.formInput}
                </Grid>
              ),
            )}
          </Grid>
        ),
      )}
      {/* eslint-enable react/no-array-index-key */}
    </>
  );
};

export default FormInputGroups;
