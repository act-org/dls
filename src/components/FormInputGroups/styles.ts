/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import makeStyles from '@mui/styles/makeStyles';

export default makeStyles(({ spacing }) => ({
  gridContainerRoot: {
    marginBottom: spacing(0.5),
  },
  gridItemRoot: {
    flex: 1,
  },
}));
