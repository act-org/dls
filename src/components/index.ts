/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

// ACTIVE COMPONENTS -----------------------------------------------------------

export * from './Alert';
export * from './AppBarNavigation';
export * from './Dialog';
export * from './EmptyState';
export * from './FormInputGroups';
export * from './FormSwitch';
export * from './Loading';
export * from './SearchBar';
export * from './SessionStorageKeySharer';
export * from './ThemeProvider';

// WIP COMPONENTS --------------------------------------------------------------

export * from './FormInput';
export * from './FormSelect';
export * from './InputLabel';
export * from './RenderMetaTags';

// DEPRECATED COMPONENTS -------------------------------------------------------

export * from './DataTable';
export * from './FormInputPrimary';
export * from './FormSelectPrimary';
export * from './InputLabelPrimary';
export * from './InputPrimary';
export * from './SelectPrimary';
export * from './SelectSecondary';
export * from './TableCellBody';
export * from './TableCellHead';
export * from './TableContainer';
