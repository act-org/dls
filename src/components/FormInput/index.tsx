/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import AlertOutline from '@actinc/dls/icons/AlertOutline';
import {
  FormControl,
  FormControlProps,
  Input,
  InputProps,
  Tooltip,
} from '@mui/material';
import { ReactElement } from 'react';

import { InputLabel, InputLabelProps } from '@actinc/dls/components/InputLabel';

export type FormInputProps = InputProps & {
  errorMessage?: string;
  formControlProps?: FormControlProps;
  helpText?: string | ReactElement;
  label?: string | ReactElement;
  labelProps?: InputLabelProps;
};

export function FormInput({
  disabled,
  error,
  errorMessage,
  helpText,
  id,
  label,
  labelProps,
  required,
  fullWidth,
  formControlProps,
  ...formInputProps
}: FormInputProps): ReactElement<FormInputProps> {
  return (
    <FormControl {...formControlProps} fullWidth={fullWidth}>
      <InputLabel
        {...labelProps}
        disabled={disabled}
        error={error}
        helpText={helpText}
        htmlFor={id}
        required={required}
      >
        {label}
      </InputLabel>
      <Input
        disabled={disabled}
        error={error}
        id={id}
        required={required}
        {...formInputProps}
      />

      {errorMessage && (
        <Tooltip arrow placement="top" title={errorMessage}>
          <AlertOutline color="error" />
        </Tooltip>
      )}
    </FormControl>
  );
}

FormInput.defaultProps = {
  errorMessage: undefined,
  formControlProps: undefined,
  helpText: undefined,
  label: undefined,
  labelProps: undefined,
};

export default FormInput;
