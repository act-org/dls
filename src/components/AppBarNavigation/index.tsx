/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { AppBarProps } from '@mui/material/AppBar';
import Grid from '@mui/material/Grid';
import { FC, ReactElement } from 'react';

import RenderLogo, { RenderLogoProps } from './RenderLogo';
import RenderNavigationItem, {
  RenderNavigationItemProps,
} from './RenderNavigationItem';
import { StyledAppBar, StyledGrid } from './styles';

export interface AppBarNavigationProps {
  appBarHeight?: number;
  AppBarProps?: AppBarProps;
  headerRightElement?: ReactElement<unknown>;
  RenderLogoProps: RenderLogoProps;
  navigationItems: RenderNavigationItemProps[];
}

export const AppBarNavigation: FC<AppBarNavigationProps> = ({
  appBarHeight,
  AppBarProps: appBarProps,
  headerRightElement,
  RenderLogoProps: logoProps,
  navigationItems,
}: AppBarNavigationProps): ReactElement<AppBarNavigationProps> => (
  <StyledAppBar
    $appBarHeight={appBarHeight || 72}
    color="transparent"
    {...appBarProps}
  >
    <RenderLogo {...logoProps} />

    <StyledGrid container spacing={4}>
      {navigationItems.map(
        (n: RenderNavigationItemProps): ReactElement<unknown> => (
          <Grid item key={n.title}>
            <RenderNavigationItem {...n} />
          </Grid>
        ),
      )}
    </StyledGrid>

    {Boolean(headerRightElement) && (
      <Grid style={{ marginLeft: 'auto' }}>{headerRightElement}</Grid>
    )}
  </StyledAppBar>
);

AppBarNavigation.defaultProps = {
  appBarHeight: undefined,
  AppBarProps: undefined,
  headerRightElement: undefined,
};

export default AppBarNavigation;
