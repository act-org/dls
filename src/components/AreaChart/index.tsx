/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

/* eslint-disable react/no-array-index-key */

import { Grid } from '@mui/material';
import { useTheme } from '@mui/material/styles';
import clsx from 'clsx';
import React from 'react';
import {
  Area,
  AreaChart as RechartsAreaChart,
  Legend,
  LegendProps,
  Tooltip,
  TooltipProps,
  ResponsiveContainer,
  ResponsiveContainerProps,
  XAxis,
  YAxis,
  XAxisProps,
  YAxisProps,
  Label,
  ReferenceLine,
  AreaProps,
  LabelProps,
} from 'recharts';
import { CategoricalChartProps } from 'recharts/types/chart/generateCategoricalChart';
import {
  ValueType,
  NameType,
} from 'recharts/types/component/DefaultTooltipContent';

export interface AreaDataProps {
  name?: string;
  [key: string]: unknown;
}

export interface AreaColorProps {
  [key: string]: string;
}

export interface AreaChartProps {
  areaChartProps?: CategoricalChartProps;
  areaKeys: string[];
  areaProps?: Partial<AreaProps>;
  children?: React.ReactNode;
  colors?: AreaColorProps;
  data: AreaDataProps[];
  height?: number;
  legendProps?: LegendProps;
  responsiveContainerProps?: ResponsiveContainerProps;
  showLegend?: boolean;
  tooltipProps?: TooltipProps<ValueType, NameType>;
  width?: number;
  xAxisProps?: XAxisProps;
  xLabel?: string;
  xLabelProps?: LabelProps;
  yAxisProps?: YAxisProps;
  yLabel?: string;
  yLabelProps?: LabelProps;
  yReferenceValue?: number;
}

/**
 * # AreaChart
 *
 */
export const AreaChart: React.FC<AreaChartProps> = ({
  children,
  colors,
  areaProps,
  areaChartProps,
  responsiveContainerProps,
  tooltipProps,
  data,
  height,
  legendProps,
  width,
  showLegend = false,
  areaKeys,
  xAxisProps,
  yAxisProps,
  xLabel,
  xLabelProps,
  yLabel,
  yLabelProps,
  yReferenceValue,
}: AreaChartProps): React.ReactElement => {
  const { customColors, palette, spacing, typography }: any = useTheme();

  return (
    <Grid container direction="column" item spacing={2} xs={12}>
      <ResponsiveContainer
        debounce={50}
        height={height || 400}
        width={width || '100%'}
        {...responsiveContainerProps}
      >
        <RechartsAreaChart
          data={data}
          margin={{
            bottom: parseInt(String(spacing(1)), 10),
            left: parseInt(String(spacing(1)), 10),
            right: parseInt(String(spacing(1)), 10),
            top: parseInt(String(spacing(1)), 10),
          }}
          style={{ fontFamily: typography.fontFamily, position: 'absolute' }}
          {...areaChartProps}
        >
          <XAxis
            axisLine={false}
            dataKey="name"
            height={50}
            orientation="bottom"
            padding={{ left: 40, right: 30 }}
            style={{
              fill: palette.secondary.dark,
              fontSize: typography.caption.fontSize,
            }}
            tickLine={false}
            xAxisId={0}
            {...xAxisProps}
          >
            {xLabel && (
              <Label
                position="insideBottom"
                style={{
                  fill: palette.secondary.dark,
                  fontSize: typography.h4.fontSize,
                  paddingTop: '10px',
                  userSelect: 'none',
                }}
                {...xLabelProps}
              >
                {xLabel}
              </Label>
            )}
          </XAxis>
          <YAxis
            axisLine={false}
            padding={{ bottom: 20, top: 20 }}
            style={{
              fill: palette.secondary.dark,
              fontSize: typography.h4.fontSize,
              fontWeight: Number(typography.fontWeightRegular),
            }}
            tickCount={5}
            tickLine={false}
            type="number"
            yAxisId={0}
            {...yAxisProps}
          >
            {yLabel && (
              <Label
                angle={-90}
                offset={15}
                position="insideLeft"
                style={{
                  fill: palette.secondary.dark,
                  fontSize: typography.h4.fontSize,
                  userSelect: 'none',
                }}
                {...yLabelProps}
              >
                {yLabel}
              </Label>
            )}
          </YAxis>
          {(yReferenceValue || yReferenceValue === 0) && (
            <ReferenceLine stroke={palette.grey[400]} y={yReferenceValue} />
          )}
          {areaKeys.map((key, i) => (
            <Area
              dataKey={key}
              fill={clsx(
                i === 0 && customColors?.chart?.primary?.first,
                i === 1 && customColors?.chart?.primary?.second,
                i === 2 && customColors?.chart?.primary?.third,
                i === 3 && customColors?.chart?.primary?.fourth,
                i === 4 && customColors?.chart?.primary?.fifth,
                i === 5 && customColors?.chart?.primary?.sixth,
                i > 5 && palette.grey[700],
              )}
              fillOpacity={1}
              key={`${key}-area`}
              stroke="none"
              style={
                colors?.[key]
                  ? { ...areaProps?.style, fill: colors[key] }
                  : areaProps?.style
              }
              {...(areaProps as Omit<AreaProps, 'dataKey' | 'ref'>)}
            />
          ))}

          {showLegend && (
            <Legend
              align={data.length > 4 ? 'right' : 'center'}
              layout="vertical"
              verticalAlign={data.length > 6 ? 'top' : 'middle'}
              {...(legendProps as Legend)}
            />
          )}
          <Tooltip
            itemStyle={{ color: palette.secondary.dark }}
            labelStyle={{ color: palette.secondary.dark }}
            wrapperStyle={{ outline: 'none' }}
            {...tooltipProps}
          />
          {children}
        </RechartsAreaChart>
      </ResponsiveContainer>
    </Grid>
  );
};

AreaChart.defaultProps = {
  areaChartProps: undefined,
  areaProps: undefined,
  children: undefined,
  colors: undefined,
  height: undefined,
  legendProps: undefined,
  responsiveContainerProps: undefined,
  showLegend: undefined,
  tooltipProps: undefined,
  width: undefined,
  xAxisProps: undefined,
  xLabel: undefined,
  xLabelProps: undefined,
  yAxisProps: undefined,
  yLabel: undefined,
  yLabelProps: undefined,
  yReferenceValue: undefined,
};

export default AreaChart;
