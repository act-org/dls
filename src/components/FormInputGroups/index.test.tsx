/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { noop } from 'lodash';

import { FormInput } from '~/components/FormInput';
import { standard } from '~/helpers/test';

import FormInputGroups from '.';

describe('FormInputGroups', () => {
  const Component = (
    <FormInputGroups
      groups={[
        [
          {
            formInput: (
              <FormInput
                inputProps={{
                  onChange: noop,
                  placeholder: 'Input 1',
                  type: 'text',
                  value: 'value1',
                }}
                labelProps={{
                  children: 'Input 1',
                }}
                required
              />
            ),
            key: 'INPUT_1',
          },
          {
            formInput: (
              <FormInput
                inputProps={{
                  onChange: noop,
                  placeholder: 'Input 2',
                  type: 'password',
                  value: 'value2',
                }}
                labelProps={{
                  children: 'Input 2',
                }}
                required
              />
            ),
            key: 'INPUT_2',
          },
        ],
        [
          {
            formInput: (
              <FormInput
                inputProps={{
                  onChange: noop,
                  placeholder: 'Input 3',
                  type: 'text',
                  value: 'value3',
                }}
                labelProps={{
                  children: 'Input 3',
                }}
              />
            ),
            key: 'INPUT_3',
          },
        ],
      ]}
    />
  );
  standard(Component);
});
