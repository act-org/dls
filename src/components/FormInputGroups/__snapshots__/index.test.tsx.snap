// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputGroups ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-1 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-3 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 1"
            required=""
            type="text"
            value="value1"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-3 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 2"
            required=""
            type="password"
            value="value2"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-1 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-3 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 3"
            type="text"
            value="value3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormInputGroups ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-4 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-5"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 1"
            required=""
            type="text"
            value="value1"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-5"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 2"
            required=""
            type="password"
            value="value2"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-4 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-5"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 3"
            type="text"
            value="value3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormInputGroups ENCOURA_DATALAB theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-7 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-8"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-9 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 1"
            required=""
            type="text"
            value="value1"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-8"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-9 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          required=""
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 2"
            required=""
            type="password"
            value="value2"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-7 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-8"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-9 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          <div
            class="label-split"
          >
            <span
              class="label-help"
            />
          </div>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Input 3"
            type="text"
            value="value3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
